UM/View/RenderBatch.py

```diff
@@ -0,0 +1,297 @@
+        LineStrip = 0x0003
+        Triangles = 0x0004
+        TriangleStrip = 0x0005
+        TriangleFan = 0x0006
+
+    ##  Init method.
+    #
+    #   \param shader The shader to use for this batch.
+    #   \param kwargs Keyword arguments.
+    #                 Possible values:
+    #                 - type: The RenderType to use for this batch. Defaults to RenderType.Solid.
+    #                 - mode: The RenderMode to use for this batch. Defaults to RenderMode.Triangles.
+    #                 - backface_cull: Whether to enable or disable backface culling. Defaults to True.
+    #                 - range: A tuple indicating the start and end of a range of triangles to render. Defaults to None.
+    def __init__(self, shader, **kwargs):
+        self._shader = shader
+        self._render_type = kwargs.get("type", self.RenderType.Solid)
+        self._render_mode = kwargs.get("mode", self.RenderMode.Triangles)
+        self._backface_cull = kwargs.get("backface_cull", True)
+        self._render_range = kwargs.get("range", None)
+        self._items = []
+
+        self._view_matrix = None
+        self._projection_matrix = None
+
+        self._gl = OpenGL.getInstance().getBindingsObject()
+
+    ##  The RenderType for this batch.
+    @property
+    def renderType(self):
+        return self._render_type
+
+    ##  The RenderMode for this batch.
+    @property
+    def renderMode(self):
+        return self._render_mode
+
+    ##  The shader for this batch.
+    @property
+    def shader(self):
+        return self._shader
+
+    ##  Whether backface culling is enabled or not.
+    @property
+    def backfaceCull(self):
+        return self._backface_cull
+
+    ##  The range of elements to render.
+    #
+    #   \return The range of elements to render, as a tuple of (start, end)
+    @property
+    def renderRange(self):
+        return self._render_range
+
+    ##  The items to render.
+    #
+    #   \return A list of tuples, where each item is (transform_matrix, mesh, extra_uniforms)
+    @property
+    def items(self):
+        return self._items
+
+    ##  Less-than comparison method.
+    #
+    #   This sorts RenderType.Solid before RenderType.Transparent
+    #   and RenderType.Transparent before RenderType.Overlay.
+    def __lt__(self, other):
+        if self._render_type == self.RenderType.Solid:
+            return True
+
+        if self._render_type == self.RenderType.Transparent and other._render_type != self.RenderType.Solid:
+            return True
+
+        return False
+
+    ##  Add an item to render to this batch.
+    #
+    #   \param transform The transformation matrix to use for rendering the item.
+    #   \param mesh The mesh to render with the transform matrix.
+    #   \param uniforms A dict of additional uniform bindings to set when rendering the item.
+    #                   Note these are set specifically for this item.
+    def addItem(self, transform, mesh, uniforms = None):
```

Comment:
Bit of nitpicking, but I would have called it transformation instead of transform (It is the transformation to be used for provided mesh after all.)